gateway/api/utils.py

```diff
@@ -448,3 +448,14 @@ def create_gpujob_allowlist():
         raise ValueError("Unable to decode gpujob allowlist") from e
 
     return gpujobs
+
+
+def sanitize_file_name(name: str | None):
+    """Sanitize name of a file"""
+    if name:
+        sanitized_name = ""
+        for c in name:
+            if c.isalnum() or c in ["_", "-", "."]:
+                sanitized_name += c
+        return sanitized_name
+    return name
```

Comment:
I usually don't like to use regex because they are not so easy to read than the original approach. Let's do one thing, I can agree with the change if every time we add a regex we add a comment explaining what the regex is doing.

Reply:
Done here: [e774e07](https://github.com/Qiskit/qiskit-serverless/pull/1547/commits/e774e07807ec446f8c4dd4e6b386a8afeb319d12)